dom.go

```diff
@@ -1612,6 +1661,14 @@ type BasicElement struct {
 	*BasicNode
 }
 
+func (e *BasicElement) Attributes() []Attribute {
+	return attributeListToAttributes(e.Get("attributes"))
+}
+
+func (e *BasicElement) Dataset() []Attribute {
+	return dataListToAttributes(e.Get("attributes"))
```

Comment:
Decided to switch the returned values for attributes to maps, that seems rather intuitive and removes the need for the extra struct and interface. But whats your thoughts on the data-set, is the support issue a no-issue or should this approach be used as is?